products.php

```diff
@@ -0,0 +1,51 @@
+<?php
+require_once 'src/Models/CategoryModel.php';
+require_once 'src/Services/ProductsDisplayService.php';
+require_once 'src/Models/ProductModel.php';
+require_once 'src/Factory/furnitureDatabaseConnector.php';
+require_once 'src/Entities/ProductsEntity.php';
+
+$db = furnitureDatabaseConnector::connect();
+define('NUM_OF_CATEGORIES', 11);
+$id = $_GET["id"];
+if (!isset($id) || !is_numeric($id) || $id > NUM_OF_CATEGORIES || $id < 1) {
+    echo '<h1>Invalid id number</h1>
+            <a href="index.php"><button>Home</button></a>';
+    return;
+} else {
+    $id = intval($id);
+}
+    $products = ProductModel::getProducts($db, $id);
+    $category = CategoryModel::getCategoryTitle($db, $id);
```

Comment:
You check if the supplied id is 'correct' (line 11), but even if it *might not be* you ask the database to find stuff with it anyway (line 18, 19) and just try to put up whatever you find?

Some checking that the supplied ID is ok is needed, but there are actually two failure modes here:
1. The id supplied is wrong data - you can check for that here.
2. The id could be correct data, but you don't get anything out of the database with it - what does your page do then? 

These two failure modes need to be handled differently.